src/shared/install.c

```diff
@@ -732,64 +728,61 @@ static int remove_marked_symlinks_fd(
                         if (!found) {
                                 _cleanup_free_ char *template = NULL;
 
-                                q = unit_name_template(de->d_name, &template);
-                                if (q < 0 && q != -EINVAL)
-                                        return q;
-                                if (q >= 0)
+                                r = unit_name_template(de->d_name, &template);
+                                if (r < 0 && r != -EINVAL)
+                                        return r;
+                                if (r >= 0)
                                         found = set_contains(remove_symlinks_to, template);
                         }
 
                         if (!found) {
-                                _cleanup_free_ char *dest = NULL;
+                                _cleanup_free_ char *dest = NULL, *dest_name = NULL;
 
-                                q = chase(p, lp->root_dir, CHASE_NONEXISTENT, &dest, NULL);
-                                if (q == -ENOENT)
+                                r = chase(p, lp->root_dir, CHASE_NONEXISTENT, &dest, NULL);
+                                if (r == -ENOENT)
                                         continue;
-                                if (q < 0) {
-                                        log_debug_errno(q, "Failed to resolve symlink \"%s\": %m", p);
-                                        install_changes_add(changes, n_changes, q, p, NULL);
```

Comment:
This is not quite only a 'modernization', before here and below the return value of install_changes_add was ignored, now it's collected and returned, so please at least mention this in the commit message if it was intentional

Reply:
Hmm, I don't actually follow? `install_changes_add` would return the original errno (`q` here), and previously it's also collected, but just separately?

Reply:
It could return the original errno, but it could also return ENOMEM or other errors which were ignored before, no?